tests/test_events/test_api.py

```diff
@@ -122,6 +137,36 @@ def test_send(self, mock_send: MagicMock) -> None:
         )
         assert self.event_api.get_all_sent_events() == [event]
 
+    @patch.object(Boto3Client, "eventbridge")
+    def test__send__sends_events_in_chunks_respecting_limits(self, mock_send: MagicMock) -> None:
+        for i in range(33):  # AWS allows sending maximum 10 events at once
+            self.event_api.register(MyTestEvent({"x": i}))
+
+        self.event_api.send()
+
+        assert mock_send.put_events.call_count == 4
+        assert len(mock_send.put_events.call_args_list[0].kwargs["Entries"]) == 10
```

Comment:
You could to it with has calls and use array multiplication so it would look like that ~:
 == ([MyTestEvent({"x": i})] * 10, [MyTestEvent({"x": i})] * 10, [MyTestEvent({"x": i})] * 3,) 
 as Event has __eq__ implement :)

Reply:
That would be true if we test here events, but this line is about entries - they are quite big dictionaries.

I did it this way because I wanted to keep it clean - the main goal is to check whether this method was triggered 4 times with a max of 10 entries per run. The correctness of building the entries is tested in other tests 😉